### ISSUE TITLE
[BUG] Cannot open visualisations in new windows or tabs

### ISSUE THREAD

**Issue by jgough** [opensearch-project/OpenSearch-Dashboards#2197]
**Describe the bug**
If I go to /app/dashboards then I can middle click or right click on a dashboard in the list and open in a new window/tab
If I go to /app/visualize then I **cannot** middle click or right click on a visualization in the list to open in a new window/tab

**To Reproduce**
1. Go to /app/visualize
2. Try to open a visualization in a new tab or window (either by middle clicking, or right clicking) and note that it is not possible

**Expected behavior**
I expected to be able to open visualizations in new tabs and windows, much like is possible for /app/dashboards.

**Screenshots**
Visualize:
![image](https://user-images.githubusercontent.com/968442/186409987-f32feae3-dd73-48e9-b522-1871a963f57a.png)
Dashboards:
![image](https://user-images.githubusercontent.com/968442/186410315-43f69092-57b2-4850-bfa1-0e78987b32f0.png)

**OpenSearch Version**
1.3.4

**Dashboards Version**
1.3.4

**Plugins**
security-dashboards-plugin

**Host/Environment (please complete the following information):**
 - Docker

**Comment by joshuarrrr** [comment]
@jgough Thanks for filing. As a fellow tab junkie, I agree this is annoying. 

I can confirm that this behavior still exists in `main` by reproducing at https://playground.opensearch.org/app/visualize

I'm pretty sure the cause of the behavior is that the visualization list items are actually `<button>` elements, although they're styled like links. For a11y and semantic purposes, it would definitely be better to use actual `<a>` links instead, in which case the browser default behavior would work as expected.

One final note is that the the dashboards listing page is still actually rendered via Angular. When we update that list rendering (potentially in https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1866), we need to make sure we don't actually introduce this same bug there.

**Comment by BSFishy** [comment]
With a quick inspect element, I can see that the issue is in fact that the element is a `button` on the visualize page instead of an `a`. Both pages seem to use a link component, so it should just be a matter of telling the component to use an `a` instead of a `button`.

I was trying to find the exact location of the issue just by browsing on Github, but I got a little lost. From what I did see, though, it looks like it may be something that needs to be changed in OUI, but I'll look more into it on Monday.

**Comment by seraphjiang** [comment]
Thanks @BSFishy for the research

@joshuarrrr  @bandinib-amzn would you help @BSFishy to locate the issue. 

cc: @AMoo-Miki  in case the fix need to change in Oui.

**Comment by BSFishy** [comment]
I've found the cause of the issue by cross-referencing the code for the dashboard page. It seems that the link for the visualize page doesn't actually have an `href` attribute:

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/77af7f9a3331d48a25631010b9e31c4c472b3266/src/plugins/visualize/public/application/utils/get_table_columns.tsx#L104-L115

Compared to the dashboard page's:

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/77af7f9a3331d48a25631010b9e31c4c472b3266/src/plugins/dashboard/public/application/listing/dashboard_listing.js#L172-L177

I'm not super familiar with OUI, but maybe we could just add the `href` as `editUrl` and then in the `onClick` handler cancel default?

**Comment by seraphjiang** [comment]
for Eui, we coud refer to https://elastic.github.io/eui/#/navigation/link

@AMoo-Miki Does Oui has equivalent doc site.

**Comment by AMoo-Miki** [comment]
> @AMoo-Miki Does Oui has equivalent doc site.

There is one coming up soon.

**Comment by BSFishy** [comment]
I had a quick conversation with @joshuarrrr, and we discussed if the `onClick` behavior could be replaced with a regular link. It basically came down to whether or not the logic in `application.navigateToApp` is trivial. Here is the relevant code:

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/a19a7412e5152c24cb046bce986c8c0772da22ea/src/core/public/application/application_service.tsx#L252-L270

As far as I'm aware, the only thing that the function is doing outside of changing the page is setting internal state, which should be done when the page is loaded through clicking the link. The `application` variable also has a method, `getUrlForApp`, to get the URL to redirect to, so I'm going to test if setting the `href` to that output and removing the `onClick` works as intended.

**Comment by joshuarrrr** [comment]
Resolved via #2395
